Bottom: 4e01eb8a1c0681cd7bcca9c1bf3502036b929d4b
Top:    79d073d6c2d4883bac75c2245a0da053def1f1d6
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-09-07 10:25:21 -0700

Refresh of dup2

---

diff --git a/src/preload.c b/src/preload.c
index 4ba38f5..fb2149b 100644
--- a/src/preload.c
+++ b/src/preload.c
@@ -978,8 +978,12 @@ int dup2(int oldfd, int newfd)
 
 	init_preload();
 	oldfdi = idm_lookup(&idm, oldfd);
-	if (oldfdi && oldfdi->type == fd_fork)
-		fork_passive(oldfd);
+	if (oldfdi) {
+		if (oldfdi->state == fd_fork_passive)
+			fork_passive(oldfd);
+		else if (oldfdi->state == fd_fork_active)
+			fork_active(oldfd);
+	}
 
 	newfdi = idm_lookup(&idm, newfd);
 	if (newfdi) {
